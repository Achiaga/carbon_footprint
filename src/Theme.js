import React from 'react';
import { ThemeProvider } from 'styled-components';
import WebFont from 'webfontloader';

const colors = {
	primary: 'white',
	secondary: 'black',
	forest: '#38a66d',
	lightForest: '#03b554',
};

const breakpoints = {
	xxl: '1180px',
	xl: '70%',
	l: '55%',
	m: '30%',
	s: '20%',
	xs: '15%',
};

const fontSizes = {
	title: '16',
	subtitle: '16',
	header: '16',
	subheader: '16',
	text: '16',
};

const fontWeights = {
	lighter: '300',
	light: '400',
	normal: '500',
	bold: '700',
};

WebFont.load({
	google: {
		families: ['Boogaloo&family=Raleway:wght@800&family=Yanone+Kaffeesatz:wght@700&display=swap'],
	},
});

const fontFamily = {
	// title: 'Boogaloo, cursive',
	title: 'Yanone Kaffeesatz, sans-serif',
	title2: 'Raleway, sans-serif',
};

const theme = {
	breakpoints,
	colors,
	fontSizes,
	fontWeights,
	fontFamily,
};

const Theme = ({ children }) => <ThemeProvider theme={theme}>{children}</ThemeProvider>;

export default Theme;
