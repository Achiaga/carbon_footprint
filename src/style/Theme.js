import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
	colors: {
		ligth: 'white',
		dark: 'black',
	},
	fonts: ['sans-serif'],
	fontSize: {
		smal: '1em',
		medium: '2em',
	},
};

const Theme = ({ children }) => (
	<ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
