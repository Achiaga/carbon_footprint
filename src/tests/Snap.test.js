import React from 'react';
import ReactDOM from 'react-dom';
import Title from '../components/Home/intro-title';
import Theme from '../Theme';

import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders correctly enzyme', () => {
	const wrapper = shallow(
		<Theme>
			<Title />
		</Theme>
	);

	expect(toJson(wrapper)).toMatchSnapshot();
});
