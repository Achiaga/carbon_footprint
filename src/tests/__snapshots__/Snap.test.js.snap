// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly enzyme 1`] = `
<ThemeProvider
  theme={
    Object {
      "breakpoints": Object {
        "l": "55%",
        "m": "30%",
        "s": 350,
        "xl": "70%",
        "xs": "15%",
        "xxl": "1180px",
      },
      "colors": Object {
        "forest": "#38a66d",
        "lightForest": "#03b554",
        "primary": "white",
        "secondary": "black",
      },
      "fontFamily": Object {
        "title": "Yanone Kaffeesatz, sans-serif",
        "title2": "Raleway, sans-serif",
      },
      "fontSizes": Object {
        "header": "16",
        "subheader": "16",
        "subtitle": "16",
        "text": "16",
        "title": "16",
      },
      "fontWeights": Object {
        "bold": "700",
        "light": "400",
        "lighter": "300",
        "normal": "500",
      },
      "query": Object {
        "bigMobileMedia": [Function],
        "mobile": [Function],
      },
    }
  }
>
  <Title />
</ThemeProvider>
`;
